hawkbit-repository/hawkbit-repository-jpa/src/test/java/org/eclipse/hawkbit/repository/jpa/RolloutManagementTest.java

```diff
@@ -161,6 +163,105 @@ public void checkRunningRolloutsDoesNotStartNextGroupIfFinishConditionIsNotHit()
     }
 
     @Test
+    @Title("Deleting targets of a rollout")
+    @Description("Verfiying that next group is started when targets of the group have been deleted.")
+    public void checkRunningRolloutsStartsNextGroupIfTargetsDeleted() {
+
+        final int amountTargetsForRollout = 15;
+        final int amountOtherTargets = 0;
+        final int amountGroups = 3;
+        final String successCondition = "100";
+        final String errorCondition = "80";
+        final Rollout createdRollout = createAndStartRollout(amountTargetsForRollout, amountOtherTargets, amountGroups,
+                successCondition, errorCondition);
+
+        finishActionAndDeleteTargetsOfFirstRunningGroup(createdRollout);
+
+        checkStatusOfRolloutGroupsAfterDeletingTargets(createdRollout);
+
+        finishActionAndDeleteTargetsOfSecondRunningGroup(createdRollout);
+
+        deleteThrirdGroup(createdRollout);
+
+        checkStatusOfRolloutAndRolloutGroupsAfterDeletingTargets(createdRollout);
+
+    }
+
+    @Step("Create a rollout by the given parameter and start it.")
+    public Rollout createAndStartRollout(final int amountTargetsForRollout, final int amountOtherTargets,
+            final int amountGroups, final String successCondition, final String errorCondition) {
+
+        final Rollout createdRollout = createSimpleTestRolloutWithTargetsAndDistributionSet(amountTargetsForRollout,
+                amountOtherTargets, amountGroups, successCondition, errorCondition);
+        rolloutManagement.startRollout(createdRollout);
+        return createdRollout;
+    }
+
+    @Step("Finish three actions of the rollout group and delete two targets")
+    public void finishActionAndDeleteTargetsOfFirstRunningGroup(final Rollout createdRollout) {
+        // finish group one by finishing targets and deleting targets
+        final List<Action> runningActions = deploymentManagement.findActionsByRolloutAndStatus(createdRollout,
+                Status.RUNNING);
+        finishAction(runningActions.get(0));
+        finishAction(runningActions.get(1));
+        finishAction(runningActions.get(2));
+        targetManagement.deleteTargets(runningActions.get(3).getTarget().getId(),
+                runningActions.get(4).getTarget().getId());
+    }
+
+    @Step("Check the status of the rollout groups")
+    public void checkStatusOfRolloutGroupsAfterDeletingTargets(final Rollout createdRollout) {
```

Comment:
Be more precicly => `checkStatusOfRolloutGroupsAfterDeletingTargets(...)` is not explaining well what this method actual does. 3 lines later follow the real propose of this method:
> validate that the second group is in running state
So name the method according to the comment and you don't need to provide an additional comment